src/Compilers/CSharp/Portable/Binder/PatternVariableFinder.cs

```diff
@@ -165,6 +165,21 @@ public override void VisitArgument(ArgumentSyntax node)
                 return;
             }
 
+            switch (node.Identifier.
+                         Parent. // ArgumentSyntax
+                         Parent. // ArgumentListSyntax
+                         Parent. // invocation/constructor initializer
+                         Kind())
+            {
+                case SyntaxKind.InvocationExpression:
+                case SyntaxKind.ObjectCreationExpression:
+                case SyntaxKind.ThisConstructorInitializer:
+                case SyntaxKind.BaseConstructorInitializer:
+                    break;
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(node.Identifier.Parent.Parent.Parent.Kind());
```

Comment:
Consider extracting a local for `node.Identifier.Parent.Parent.Parent.Kind()`, if for no other reason so the two uses of that expression cannot get out of sync.